core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,184 @@
+package ca.jrvs.apps.grep;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.log4j.BasicConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class JavaGrepImp implements JavaGrep {
+
+  final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+
+  private String regex;
+  private String rootPath;
+  private String outFile;
+
+  public static void main(String[] args) {
+    if (args.length < 3) {
+      throw new IllegalArgumentException("USAGE: JavaGrep regex rootPath outFile");
+    }
+
+    //Use default logger config
+    BasicConfigurator.configure();
+
+    JavaGrepImp javaGrepImp = new JavaGrepImp();
+    javaGrepImp.setRegex(args[0]);
+    javaGrepImp.setRootPath(args[1]);
+    javaGrepImp.setOutFile(args[2]);
+
+    try {
+      javaGrepImp.process();
+    } catch (Exception ex) {
+      javaGrepImp.logger.error(ex.getMessage(), ex);
```

Comment:
javaGrepImp.logger.error("meaningful app specific message", ex);

the ex.message() already included in the second arg. you dont' want to double log the same message.